src/CollectorRegistry.ts

```diff
@@ -20,19 +21,38 @@ export class CollectorRegistry {
     }
   }
 
+  setDefaultLabels(labelMap: Record<string, string>) {
+    this.defaultLabelMap = labelMap
+  }
+
   /**
    * Expose samples as text format
    * https://prometheus.io/docs/instrumenting/exposition_formats/#text-format-details
    */
   expose() {
     const collectors = Object.values(this.collectors)
+    const defaultLabelNames = Object.keys(this.defaultLabelMap);
     const results = []
 
     for (const collector of collectors) {
       const samples = collector.collect()
       if (samples.length > 0) {
         const help = `# HELP ${collector.name} ${collector.help}\n`
         const type = `# TYPE ${collector.name} ${collector.type}\n`
+
+        for (const sample of samples) {
+          sample.labels = sample.labels || {}
+
+          if (defaultLabelNames.length > 0) {
+            // Make a copy before mutating
+            sample.labels = { ...sample.labels }
+
+            for (const labelName of defaultLabelNames) {
+              sample.labels[labelName] = sample.labels[labelName] || this.defaultLabelMap[labelName]
+            }
+          }
```

Comment:
It looks to me like this piece of code can simply be:
```typescript
sample.labels = { ...this.defaultLabelMap, ...sample.labels }
```

Reply:
+